### ISSUE TITLE
Exclude user's email address from CC list when replying to emails

### ISSUE THREAD

**Issue by arkadiyasuratov** [espocrm/espocrm#3102]
**Describe the bug**

When you reply to an email ~from a Group Email Account~ where you're CC'd, your own email address gets added to the CC list again, even though it's already correctly set in the "From" field.

**Steps to reproduce the behavior:**

1. Set up EspoCRM with a Group Email Account and a Personal Email Account.
2. Send an email:
   - Address it to the Group Email Account (in the "TO" field).
   - Include the user's Personal Email Account in the CC list.
3. Log in as the user associated with the Personal Email Account.
4. Reply to the email.
5. Notice that in the reply:
   - The "FROM" address correctly shows the Personal Email Account.
   - However, the CC list still includes the user's own email address from the Personal Email Account, despite being the sender.

This issue arises because the system doesn't filter out the sender's own email address from the CC list when generating the reply, leading to redundancy in email recipients.

**Expected behavior**

When a user replies to an email:
- The "From" address should correctly reflect the user's email address.
- The CC list should not include the user's own email address, even if it was included in the original email's CC list.

**EspoCRM version**
8.3.5

**Possible solutions**

1. Modify the code to filter out the user's primary email address from the CC list. <https://github.com/espocrm/espocrm/blob/8.3.5/client/src/email-helper.js#L157>

```js
attributes.cc = (model.get('cc') || '').split(';').filter(item => {
    return item !== this.getUser().get('emailAddress');
});
```

2. Ensure that an email address in the "CC" list cannot be the same as the email address in the "FROM" field.

**Comment by buzzard10** [comment]
I have similar issue

**Comment by Kharg** [comment]
Salesforce adds your own email in bcc so that you can get a copy, but this can be disabled per user.

**Comment by yurikuzn** [comment]
Is the personal email address is non-primary user's? Does the issue happen for primary address?

**Comment by yurikuzn** [comment]
https://github.com/espocrm/espocrm/blob/0ffe39cec80d475e6d307b1b82d8bed786bc1911/client/src/email-helper.js#L253

From address can be automatically set to non primary address. Should we filter out the primary from cc then? It's convoluted. 

I'd call it an issue, not a bug, as it is a matter of choosing what would work the best among options.

**Comment by arkadiyasuratov** [comment]
Sure, let's simplify this:

1. **Email Setup**:
    - An email is sent to **contact@example.com**.
    - The email is also CC'd to **arkadiy@example.com**.
    - You have a personal account with **arkadiy@example.com**.
    - On your user page, **arkadiy@example.com** is your primary and only email address.

2. **Replying to the Email**:
    - When you click "Reply":
        - The "From" field automatically shows **arkadiy@example.com**.
        - The CC field also includes **arkadiy@example.com**.

In short, when you reply, your email address appears in both the "From" and "CC" fields.

You can test how this works in Gmail or Outlook. When you're CC'd and you click on "Reply to all," the email you're responding from does not appear in the CC list.

**Comment by arkadiyasuratov** [comment]
https://github.com/espocrm/espocrm/blob/0ffe39cec80d475e6d307b1b82d8bed786bc1911/client/src/email-helper.js#L248-L257

What about setting the 'from' attribute first and then setting 'cc'? That way, we can exclude the email that is set for 'from'.

**Comment by yurikuzn** [comment]
The question is only about cases when a user will change the from to another (personal or group). Would loosing CC be not desirable then?

**Comment by arkadiyasuratov** [comment]
The suggested behavior is already functioning correctly for the "To" field. If an email is sent to multiple addresses contact@example.com and arkadiy@example.com and I reply as arkadiy@example.com, then only contact@example.com is added to the CC.

https://github.com/espocrm/espocrm/blob/0ffe39cec80d475e6d307b1b82d8bed786bc1911/client/src/email-helper.js#L165-L167

**Comment by yurikuzn** [comment]
The behavior with TO and the primary address would not work well if an email will be replied from not a primary address. That address will be added to CC. The question is about multiple FROMs that can either be automatically pre-filled, or changed intentionally.

**Comment by Kharg** [comment]
> The question is only about cases when a user will change the from to another (personal or group). Would loosing CC be not desirable then?

Have the user in BCC with bool option in the user's preferences?

So the user doesn't risk of leaking his own email address if using a group email such as helpdesk.

At least this is what SF does.

**Comment by yurikuzn** [comment]
> Have the user in BCC with bool option in the user's preferences?

Don't we have this feature already? I forgot.

**Comment by Kharg** [comment]
> > Have the user in BCC with bool option in the user's preferences?
> 
> 
> 
> Don't we have this feature already? I forgot.
> 
> 
> 
> EDIT. Seems not.

If this ever gets included maybe to have a smoother transition email fields (to, cc, bcc) could behave more like arrays/multi-enum meaning that you can grab them to move them so that a recipient can be easily moved between to/cc/bcc fields?

**Comment by yurikuzn** [comment]
There are the following options what we could exclude from CC when replying:

* only user's primary address
* all user's email addresses
* only address that is preselected in From
* all user's plus all shared group addresses

**Comment by emillod** [comment]
In my opinion, all user-related email addresses should be excluded. However, given the numerous options, this could be a configurable parameter for the admin to decide.

**Comment by rabiibrahimi** [comment]
> * only address that is preselected in From

i think this make sense, hence if there are any other emails of the user than it is a choice and those emails will hit different inbox from the current one.

**Comment by yurikuzn** [comment]
Excluding personal addresses from CC: https://github.com/espocrm/espocrm/commit/668233a97802e5081cec75a45b8ccc4ed5f94322
